freqtrade/optimize/backtesting.py

```diff
@@ -457,13 +455,21 @@ def backtest_one_strategy(self, strat: IStrategy, data: Dict[str, Any], timerang
         preprocessed = self.strategy.ohlcvdata_to_dataframe(data)
 
         # Trim startup period from analyzed dataframe
-        for pair, df in preprocessed.items():
-            preprocessed[pair] = trim_dataframe(df, timerange,
-                                                startup_candles=self.required_startup)
-        min_date, max_date = history.get_timerange(preprocessed)
-        if min_date is NaT or max_date is NaT:
+        for pair in list(preprocessed.keys()):
```

Comment:
Actually, this is even, better, and enough. It doesn't create a keys view object. Sorry, should have mentioned on the initial review.
```suggestion
        for pair in list(preprocessed):
```